docs/CustomUIA.md

```diff
@@ -0,0 +1,125 @@
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT Licence. -->
+
+## Custom UI Automation properties
+Axe Windows supports the inspection of [custom UI Automation properties](https://docs.microsoft.com/en-gb/windows/win32/winauto/uiauto-custompropertieseventscontrolpatterns). The custom properties for which Axe Windows should register are described in a configuration file that is loaded once per session and applies to all inspected UI Automation elements.
+
+### Configuration file format
+Custom UI Automation configuration data is specified in a [JSON](https://en.wikipedia.org/wiki/JSON#Syntax) formatted text file in the configuration directory. The configuration file consists of one object with the following possible attributes:
+
+#### metadata
+This attribute, if specified, can contain values to describe the file as determined by the user. It may be useful, for instance, to store version or authorship information. Example:
+
+```
+  "metadata": {
+    "version": "1.0",
+    "modified": "1970-01-01"
+  }
+```
+
+#### properties
+This attribute contains an array of objects, one per defined custom UI Automation property. Each property object contains the following fields:
+
+##### guid
+The globally unique identifier of this property.
+
+##### programmaticName
+A textual description of this property.
+
+##### uiaType
+The data type of this property's value, one of `"string"`, `"int"`, or `"bool"`.
```

Comment:
Excel and PowerPoint currently only use string, int, or bool, but the [docs](https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-designingcustompropseventpatterns) also allow support of double, element, or point types. We should support the full set of supported types

Reply:
How might we specify "element" or "point" as a string/in the serialized representation?

Reply:
They'd both need to be converted to a string, similar to how we handle a BoundingRectangle. For a point, we'd just report x and y. For an element, we'd probably report the element's `Glimpse` property, which is essentially a concatenation of the element type and the element name. It's admittedly a bit crude, but it should be adequate to validate that the custom property is working as expected